lib/mayaUsd/ufe/UsdAttribute.cpp

```diff
@@ -706,137 +432,59 @@ bool UsdAttribute::_hasMetadata(const std::string& key) const
 bool UsdAttribute::hasMetadata(const std::string& key) const
 #endif
 {
-    bool result = false;
-    if (isValid()) {
-        // Special cases for known Ufe metadata keys.
-        if (key == Ufe::Attribute::kLocked) {
-            result = fUsdAttr.HasMetadata(MayaUsdMetadata->Lock);
-            if (result) {
-                return true;
-            }
-        }
-        PXR_NS::TfToken tok(key);
-        result = fUsdAttr.HasMetadata(tok);
-        if (result) {
-            return true;
-        }
-    }
-#ifdef UFE_V4_FEATURES_AVAILABLE
-#if (UFE_PREVIEW_VERSION_NUM >= 4008)
-    if (fAttrDef && fAttrDef->hasMetadata(key)) {
-        return true;
-    }
-#endif
-#endif
-    return false;
+    return _attrHolder->hasMetadata(key);
 }
 #endif
 
-#ifdef UFE_V4_FEATURES_AVAILABLE
-PXR_NS::SdfValueTypeName UsdAttribute::usdAttributeType() const { return ufeTypeToUsd(typeName()); }
-#endif
+PXR_NS::SdfValueTypeName UsdAttribute::usdAttributeType() const
+{
```

Comment:
New function to return the USD attribute type (the type we should use when creating USD attributes).